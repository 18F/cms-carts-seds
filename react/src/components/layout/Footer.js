import React, { Component } from "react";

class Footer extends Component {
  render() {
    return (
      <div className="footer">
        <div className="adverts">
          <div className="ds-l-container">
            <div class="ds-l-row">
              <div class="chip-info ds-l-col--6">
                <img src="/img/new-york-temp.png" alt="Medicaid.gov" />
                <img src="/img/new-york-temp.png" alt="MACPro.gov" />
              </div>
              <div className="cms-branding ds-l-col--6">
                <div class="cms-logo">
                  <img
                    src="/img/new-york-temp.png"
                    alt="Department of Health and Human Services, USA"
                  />
                </div>
                <div class="cms-copy">
                  A federal government website managed and paid for by the U.S.
                  Centers for Medicare and Medicaid Services and part of the
                  MACPro suite.
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="info">
          <div className="ds-l-container">
            <div class="ds-l-row">
              <div class="help ds-l-col--6">
                Email{" "}
                <a href="mailto:cartshelp@cms.hhs.gov">CARTSHELP@cms.hhs.gov</a>{" "}
                for help or feedback.
              </div>
              <div className="address ds-l-col--6">
                7500 Security Boulevard Baltimore, MD 21244
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Footer;
