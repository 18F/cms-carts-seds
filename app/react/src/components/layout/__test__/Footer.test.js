import React from "react";
import Footer from "../Footer";

import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";

describe("Footer Component", () => {
  const component = shallow(<Footer />);

  it("renders", () => {
    expect(component.exists()).toBe(true);
  });

  it("has appropriate classnames", () => {
    expect(component.exists(".footer")).toBe(true);
  });

  it("includes federal website disclaimer", () => {
    expect(
      component.find('[href="mailto:cartshelp@cms.hhs.gov"]').hasClass("help")
    ).toBe(true);
  });
});

describe("Footer Component, snapshot testing", () => {
  const snapShot = renderer.create(<Footer />);

  it("should match the snapshot", () => {
    expect(snapShot.toJSON()).toMatchSnapshot();
  });
});
