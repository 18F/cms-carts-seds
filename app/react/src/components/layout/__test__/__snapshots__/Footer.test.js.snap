// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component, snapshot testing should match the snapshot 1`] = `
<div
  className="footer"
>
  <div
    className="adverts"
  >
    <div
      className="ds-l-container"
    >
      <div
        className="ds-l-row"
      >
        <div
          className="chip-info ds-l-col--6"
        >
          <div
            className="medicaid-logo"
          >
            <img
              alt="Medicaid.gov"
              src="/img/medicadelogo.png"
            />
          </div>
          <div
            className="macpro-logo"
          >
            <img
              alt="MACPro.gov"
              src="/img/MACProLogotransparent.png"
            />
          </div>
        </div>
        <div
          className="cms-branding ds-l-col--6"
        >
          <div
            className="cms-logo"
          >
            <img
              alt="Department of Health and Human Services, USA"
              src="/img/hhs-logo.png"
            />
          </div>
          <div
            className="cms-copy"
          >
            A federal government website managed and paid for by the U.S. Centers for Medicare and Medicaid Services and part of the MACPro suite.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="info"
  >
    <div
      className="ds-l-container"
    >
      <div
        className="ds-l-row"
      >
        <div
          className="help ds-l-col--6"
        >
          Email
           
          <a
            className="help"
            href="mailto:cartshelp@cms.hhs.gov"
          >
            CARTSHELP@cms.hhs.gov
          </a>
           
          for help or feedback.
        </div>
        <div
          className="address ds-l-col--6"
        >
          7500 Security Boulevard Baltimore, MD 21244
        </div>
      </div>
    </div>
  </div>
</div>
`;
